File: samplePython/sample.py
def greet(name):
    print(f"Hello, {name}!")


def add(a, b):
    return a + b


class Calculator:
    def __init__(self):
        self.memory = 0

    def multiply(self, x, y):
        result = x * y
        self.memory = result
        return result

    @staticmethod
    def subtract(x, y):
        return x - Y


greet("Morgan")

sum_result = add(10, 5)
print(f"Sum: {sum_result}")

calc = Calculator()

product = calc.multiply(3, 4)
print(f"Product: {product}")
print(f"Stored in memory: {calc.memory}")

difference = Calculator.subtract(9, 2)
print(f"Difference: {difference}")

File: sampleCode/sample.cpp
#include <iostream>
using namespace std;

// Top-level function
void greet(const string& name) {
    cout << "Hello, " << name << "!" << endl;
}

// A class with member functions
class Calculator {
public:
    Calculator() : memory(0) {}

    int add(int a, int b) {
        memory = a + b;
        return memory;
    }

    int multiply(int a, int b) {
        memory = a * b;
        return memory;
    }


private:
    int memory;

};

// A struct with a method
struct Logger {
    void log(const string& msg) {
        cout << "[LOG] " << msg << endl;
    }
};

int main() {
    greet("Morgan");

    Calculator calc;
    cout << "Sum: " << calc.add(2, 3) << endl;

    Logger logger;
    logger.log("Calculation complete");

    return 0;
}